pkg/registry/apiserverinternal/storageversion/strategy.go

```diff
@@ -96,8 +103,71 @@ func (storageVersionStatusStrategy) PrepareForUpdate(ctx context.Context, obj, o
 	metav1.ResetObjectMetaForStatus(&newSV.ObjectMeta, &oldSV.ObjectMeta)
+
+	status := apiserverinternal.ConditionFalse
+	commonEncodingVersion := "<unset>"
+	reason := commonEncodingVersionUnsetReason
+	if newSV.Status.CommonEncodingVersion != nil {
+		status = apiserverinternal.ConditionTrue
+		commonEncodingVersion = *newSV.Status.CommonEncodingVersion
+		reason = commonEncodingVersionSetReason
+	}
+	condition := apiserverinternal.StorageVersionCondition{
+		Type:   apiserverinternal.AllEncodingVersionsEqual,
+		Status: status,
+		// Generation corresponds to the desired state, which is always
+		// 0 for storage versions, because StorageVersionSpec is empty.
+		ObservedGeneration: newSV.Generation,
+		Reason:             reason,
+		Message:            fmt.Sprintf("The common encoding version is: %s", commonEncodingVersion),
+	}
+	setStatusCondition(&newSV.Status.Conditions, condition)
 }
 
 func (storageVersionStatusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
 	return validation.ValidateStorageVersionStatusUpdate(obj.(*apiserverinternal.StorageVersion), old.(*apiserverinternal.StorageVersion))
 }
+
+// setStatusCondition sets the corresponding condition in conditions to newCondition.
+// conditions must be non-nil.
+// 1. if the condition of the specified type already exists (all fields of the existing condition are updated to
+//    newCondition, LastTransitionTime is set to now if the new status differs from the old status)
+// 2. if a condition of the specified type does not exist (LastTransitionTime is set to now() if unset,
+// and newCondition is appended)
+func setStatusCondition(conditions *[]apiserverinternal.StorageVersionCondition, newCondition apiserverinternal.StorageVersionCondition) {
+	if conditions == nil {
+		return
+	}
+	existingCondition := findStatusCondition(*conditions, newCondition.Type)
+	if existingCondition == nil {
+		if newCondition.LastTransitionTime.IsZero() {
+			newCondition.LastTransitionTime = metav1.NewTime(time.Now())
+		}
+		*conditions = append(*conditions, newCondition)
+		return
+	}
+
+	if existingCondition.Status != newCondition.Status {
+		existingCondition.Status = newCondition.Status
+		if !newCondition.LastTransitionTime.IsZero() {
+			existingCondition.LastTransitionTime = newCondition.LastTransitionTime
+		} else {
+			existingCondition.LastTransitionTime = metav1.NewTime(time.Now())
+		}
+	}
+
+	existingCondition.Reason = newCondition.Reason
+	existingCondition.Message = newCondition.Message
+	existingCondition.ObservedGeneration = newCondition.ObservedGeneration
+}
+
+// findStatusCondition finds the conditionType in conditions.
+func findStatusCondition(conditions []apiserverinternal.StorageVersionCondition, conditionType apiserverinternal.StorageVersionConditionType) *apiserverinternal.StorageVersionCondition {
+	for i := range conditions {
+		if conditions[i].Type == conditionType {
+			return &conditions[i]
+		}
+	}
+
+	return nil
+}
```

Comment:
We need unit tests.

Reply:
will move the strategy change to a separate PR